Lion: `2.2.6` & `2.2.8`

sf-dcp opened the pull request:
Related to #1571

Successful [build](https://github.com/NYCPlanning/data-engineering/actions/runs/14851255477/job/41695181946)

This PR adds 2 fields to the `lion` table. 

For the most part, the logic between the 2 fields is very similar: it's based on a segment's atomic polygons. For `2.2.6`, commit with part || covers the case when a segment is within same atomic polygon and the output value depends on whether the segment is a dead-end (There are no other segments attached to it). 

I added the tables in red:

<img width="1008" alt="image" src="https://github.com/user-attachments/assets/a1a98f06-5b6f-41fa-8698-693afc385c12" />

### Testing data

Please refer to the issue. There is a mismatch for the `Borough Boundary Indicator` field due to underlying atomic polygons.

fvankrieken commented on products/lion/models/intermediate/2.2.6/int__centerline_segment_locational_status.sql:
```diff
@@ -0,0 +1,116 @@
+    SELECT
+        *,
+        CASE
+            WHEN centerline_borocode <> left_borocode AND centerline_borocode = right_borocode THEN 'L'
```
For these `WHEN`s, we're already specifying that left <> right in the filter. So we can just check one of these two conditions, the other is guaranteed to be true.

This goes for all four `WHEN` clauses in this CTE. I get maybe being explicit about the conditions, but I think given how complicated this query is, keeping the CASE WHENs simpler based on the filter conditions makes it easier to read and digest

fvankrieken commented on products/lion/models/intermediate/2.2.6/int__centerline_segment_locational_status.sql:
```diff
@@ -0,0 +1,116 @@
+)
+
+SELECT * FROM left_city_boundary_aps
+UNION
```
Again, let's use UNION ALL

sf-dcp commented on products/lion/models/intermediate/2.2.6/int__centerline_segment_locational_status.sql:
```diff
@@ -0,0 +1,116 @@
+        CASE
+            WHEN
+                centerline_borocode <> left_borocode AND centerline_borocode = right_borocode
+                THEN left_borocode::char(1)
```
I believe this value gets casted to string here by default anyways because the `segment_locational_status` column has non-numeric values as well. So, I either casted to be explicit or because the query wasn't working

sf-dcp commented on products/lion/models/intermediate/2.2.6/_int_226.yml:
Yeah, I tried to do one big table but it was hard to parse the logic. I will add the test to cross-check record count with the atomic polygons table (the underlying table that gets chunked up)

sf-dcp commented:
@fvankrieken I added all the suggestions and updated the GHA to run `dbt build` for all tables. Successful run [here](https://github.com/NYCPlanning/data-engineering/actions/runs/14865272150/job/41740403330). Interestingly, the build ran only 5 minutes! Maybe building entire thing without separating dbt commands expedites the run...

fvankrieken commented on products/lion/models/intermediate/2.2.6/int__centerline_segment_locational_status.sql:
```diff
@@ -67,16 +67,14 @@ different_aps_different_boros AS (
         CASE
             WHEN centerline_borocode <> left_borocode AND centerline_borocode = right_borocode THEN 'L'
-            WHEN centerline_borocode = left_borocode AND centerline_borocode <> right_borocode THEN 'R'
+            ELSE 'R'
```
Sorry, this is not entirely what I meant - I meant within the same WHEN clause

Given that we have a clause filtering down WHERE left <> right (boro), all we need for one of these clauses is

```
CASE
    WHEN centerline_borocode <> left_borocode THEN 'L'
    WHEN centerline_borocode <> right_borocode THEN 'R'
END AS borough_boundary_indicator
```

and then something similar for seg loc status

To your code, the second `WHEN` clause is somewhat unnecessary as well, but I think it helps with understanding the logic to leave it. But my point was more around readability and flow of logic of these individual `WHEN` clauses
- we're in the CTE where left boro <> right boro
- in those cases, we are assigning the boundary indicator to the side whose borough differs from centerline

fvankrieken replied:
I guess I'm missing the case where neither left or right boro equals the centerline boro though

fvankrieken replied:
So then maybe

```
CASE
    WHEN centerline_borocode = right_borocode THEN 'L'
    WHEN centerline_borocode = left_borocode THEN 'R'
END AS borough_boundary_indicator
```

fvankrieken replied:
Let's just drop this post-review commit for now. I do think this could be cleaned up slightly but to your point we maybe should have more upstream tests in that sort of case.

Sorry about this

sf-dcp replied:
Per teams chat, I revised the logic.

sf-dcp commented:
> Assuming the pr test doesn't suddenly freak out for no reason!

Yes, re-ran the build and everything passed! Yeah, it's def faster with using the dbt build command... Took only 6 min for the whole run where the build is 3 minutes.
